### ISSUE TITLE
Error upgrade from  0.19.4~ynh3 to 0.19.4~ynh4

### ISSUE THREAD

**Issue by MediaElf** [YunoHost-Apps/kresus_ynh#170]
Hi there,
I can't upgrade kresus from 0.19.4 ynh3 to 0.19.4 ynh4

- Hardware: Other ARM board 
- YunoHost version: 11.2.5 
- I have access to my server: Through SSH | through the webadmin 
- My YunoHost instance tweaked ?: no

### Steps to reproduce

- I performed a command from the CLI as root :
  yunohost app upgrade kresus -> ERROR

- I used also the webadmin:
- Just click on UPGRADE button -> ERROR

### Logs

root@younohost:~# 
yunohost app upgrade kresus

Info: Now upgrading kresus...
Info: [+...................] > Loading installation settings...
Info: [#+..................] > Checking version...
Info: [##+.................] > Backing up the app before upgrading (may take a while)...
Info: [###+................] > Stopping a systemd service...
Info: The service kresus has correctly executed the action stop.
Info: [####+...............] > Ensuring downward compatibility...
Info: [#####+..............] > Making sure dedicated system user exists...
Info: [######+.............] > Upgrading dependencies...
Info: [#######+............] > Upgrading NGINX web server configuration...
Info: [########+...........] > Installing woob...
Info: [#########+..........] > Installing app...
Warning: [Error] Upgrade failed.
Warning: 366723 Could not restore kresus: An error occured inside the app restore script
Warning: 366801 Here's an extract of the logs before the crash. It might help debugging the error:
Warning: 372326 Database kresus not found
Warning: 372796 User kresus not found
Warning: 376976 sed: can't read /opt/node_n/ynh_app_version: No such file or directory
Warning: 376990 grep: /opt/node_n/ynh_app_version: No such file or directory
Warning: 403977 The operation 'Restore 'kresus' from a backup archive' could not be completed. Please share the full log of this operation using the command 'yunohost log share 20231026-180719-backup_restore_app-kresus' to get help
Warning: 404700 Nothing was restored
Warning: Uhoh ... Yunohost failed to restore the app to the way it was before the failed upgrade :|
Error: Could not upgrade kresus: An error occurred inside the app upgrade script
Info: The operation 'Upgrade the 'kresus' app' could not be completed. Please share the full log of this operation using the command 'yunohost log share 20231026-175352-app_upgrade-kresus' to get help
Warning: Here's an extract of the logs before the crash. It might help debugging the error:
Info: DEBUG - 366805 DEBUG - + sed --in-place 's^N_PREFIX=${N_PREFIX-.*}$N_PREFIX=${N_PREFIX-/opt/node_n}g' /opt/node_n/bin/n
Info: DEBUG - 366807 DEBUG - + PATH=/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/node_n/n/versions/node/16/bin:/opt/node_n/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
Info: DEBUG - 366815 DEBUG - + test -x /usr/bin/node_n
Info: DEBUG - 366817 DEBUG - + test -x /usr/bin/npm_n
Info: DEBUG - 366818 DEBUG - ++ uname --machine
Info: DEBUG - 366820 DEBUG - + uname=aarch64
Info: DEBUG - 366821 DEBUG - + [[ aarch64 =~ aarch64 ]]
Info: DEBUG - 366823 DEBUG - + n 16 --arch=arm64
Info: DEBUG - 366825 DEBUG -   installing : node-v16.20.2
Info: DEBUG - 366826 DEBUG -        mkdir : /opt/node_n/n/versions/node/16.20.2
Info: DEBUG - 366828 DEBUG -        fetch : https://nodejs.org/dist/v16.20.2/node-v16.20.2-linux-arm64.tar.xz
Info: DEBUG - 366829 DEBUG - curl: (18) transfer closed with 17832692 bytes remaining to read
Info: DEBUG - 366831 DEBUG - xz: (stdin): Unexpected end of input
Info: DEBUG - 366832 DEBUG - tar: Unexpected EOF in archive
Info: DEBUG - 366833 DEBUG - tar: Unexpected EOF in archive
Info: DEBUG - 366835 DEBUG - tar: Error is not recoverable: exiting now
Info: DEBUG - 366837 DEBUG -
Info: DEBUG - 366839 DEBUG -   Error: failed to download archive for 16.20.2
Info: DEBUG - 366840 DEBUG -
Info: DEBUG - 366842 DEBUG - + ynh_exit_properly
Error: The operation 'Upgrade the 'kresus' app' could not be completed. Please share the full log of this operation using the command 'yunohost log share 20231026-175352-app_upgrade-kresus' to get help

FULL LOG:
https://paste.yunohost.org/raw/itekinugiv

Thank you for your help :+1:

**Comment by MediaElf** [comment]
> Hi, > > > Your system failed to decompress the node archive. Maybe it was corrupted for a while or there was a memory issue. Can you try again?

I uninstalled the container with all the data and settings, but now I can't reinstall it.

**Comment by nicofrand** [comment]
What container! Yunohost does not use containers afaik.
Can you give the full logs please?

**Comment by nicofrand** [comment]
```
2023-10-26 18:08:29,338: DEBUG - 98819   installing : node-v16.20.2
2023-10-26 18:08:29,936: DEBUG - 99416        mkdir : /opt/node_n/n/versions/node/16.20.2
2023-10-26 18:08:29,951: DEBUG - 99431        fetch : https://nodejs.org/dist/v16.20.2/node-v16.20.2-linux-arm64.tar.xz
2023-10-26 18:12:55,638: DEBUG - 365119 curl: (18) transfer closed with 17832692 bytes remaining to read
```

Can you try 

```
curl https://nodejs.org/dist/v16.20.2/node-v16.20.2-linux-arm64.tar.xz -o /tmp/node.xz
sha256sum /tmp/node.xz
```
